python/cudf_polars/cudf_polars/streaming/io.py

```
@@ -580,16 +568,81 @@ class StreamingScan(IR):
 
+    @classmethod
+    def for_split_files(
+        cls,
+        base_scan: Scan,
+        plan: IOPartitionPlan,
+        partition_count: int,
+        *,
+        rank: int,
+        nranks: int,
+        parquet_options: ParquetOptions,
+    ) -> Self:
```

Ruff (0.15.15)

[error] Undefined name `Self`

(F821)